benchmarks/src/main.rs

```diff
@@ -141,304 +193,334 @@ const WRITE_BENCHES: &[&'static str] = &[
-        };
-        println!("Running {}...", FLAT_BSON_DECODING);
-        let bson_flat_decode =
-            bench::run_benchmark::<BsonDecodeBenchmark>(bson_flat_decode_options).await?;
-
-        comp_score += score_test(bson_flat_decode, FLAT_BSON_DECODING, 75.31, more_info);
-    }
-
-    // BSON flat document encode
-    if ids[10] {
-        let bson_flat_encode_options = bench::bson_encode::Options {
-            num_iter: 10_000,
-            path: DATA_PATH.join("extended_bson").join("flat_bson.json"),
-        };
-        println!("Running {}...", FLAT_BSON_ENCODING);
-        let bson_flat_encode =
-            bench::run_benchmark::<BsonEncodeBenchmark>(bson_flat_encode_options).await?;
-
-        comp_score += score_test(bson_flat_encode, FLAT_BSON_ENCODING, 75.31, more_info);
-    }
-
-    // BSON deep document decode
-    if ids[11] {
-        let bson_deep_decode_options = bench::bson_decode::Options {
-            num_iter: 10_000,
-            path: DATA_PATH.join("extended_bson").join("deep_bson.json"),
-        };
-        println!("Running {}...", DEEP_BSON_DECODING);
-        let bson_deep_decode =
-            bench::run_benchmark::<BsonDecodeBenchmark>(bson_deep_decode_options).await?;
-
-        comp_score += score_test(bson_deep_decode, DEEP_BSON_DECODING, 19.64, more_info);
-    }
-
-    // BSON deep document encode
-    if ids[12] {
-        let bson_deep_encode_options = bench::bson_encode::Options {
-            num_iter: 10_000,
-            path: DATA_PATH.join("extended_bson").join("deep_bson.json"),
-        };
-        println!("Running {}...", DEEP_BSON_ENCODING);
-        let bson_deep_encode =
-            bench::run_benchmark::<BsonEncodeBenchmark>(bson_deep_encode_options).await?;
-
-        comp_score += score_test(bson_deep_encode, DEEP_BSON_ENCODING, 19.64, more_info);
-    }
-
-    // BSON full document decode
-    if ids[13] {
-        let bson_full_decode_options = bench::bson_decode::Options {
-            num_iter: 10_000,
-            path: DATA_PATH.join("extended_bson").join("full_bson.json"),
-        };
-        println!("Running {}...", FULL_BSON_DECODING);
-        let bson_full_decode =
-            bench::run_benchmark::<BsonDecodeBenchmark>(bson_full_decode_options).await?;
-
-        comp_score += score_test(bson_full_decode, FULL_BSON_DECODING, 57.34, more_info);
-    }
-
-    // BSON full document encode
-    if ids[14] {
-        let bson_full_encode_options = bench::bson_encode::Options {
-            num_iter: 10_000,
-            path: DATA_PATH.join("extended_bson").join("full_bson.json"),
-        };
-        println!("Running {}...", FULL_BSON_ENCODING);
-        let bson_full_encode =
-            bench::run_benchmark::<BsonEncodeBenchmark>(bson_full_encode_options).await?;
-
-        comp_score += score_test(bson_full_encode, FULL_BSON_ENCODING, 57.34, more_info);
+    let mut sorted_ids: Vec<BenchmarkId> = ids.into_iter().collect();
+    sorted_ids.sort();
```

Comment:
It might be worth randomizing the id order rather than sorting to eliminate accidental cache priming effects and the like.

Reply:
I think the ergonomics of having a consistent order for when doing manual benchmarking / debugging outweigh the potential risks of cache priming, though that's in part because I don't have a ton of knowledge as to how severe these effects can be. What do you think?

Reply:
I don't have a whole lot of knowledge here myself, so if there's good concrete benefit to the sorting keeping it that way sounds fine.